lib/src/modules/mod.rs

```diff
@@ -256,4 +256,12 @@ pub mod mods {
         info.lnk = protobuf::MessageField(invoke::<Lnk>(data));
         info
     }
+
+    /// A vector of all module names. Useful for displaying currently compiled
+    /// modules.
+    ///
+    /// See the "debug modules" command.
+    pub fn module_names() -> Vec<&'static str> {
+        super::BUILTIN_MODULES.keys().map(|&k| k).collect()
```

Comment:
Return an iterator instead of a vector, so that allocating the vector is not always required.

Reply:
Done. I was trying to avoid it because the return type now exposes the Module type, which means I had to alter it's visibility. It didn't feel right to do that, so please let me know if there is a better way!

Reply:
Ah, I think I am starting to understand more. Thanks!